File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    if s[0] == ')' or s[len(s) - 1] == '(': return False;

    stack = [];
    for i in s:
        if i == '(': stack.append(i);
        elif len(stack) != 0 and i == ')': stack.pop();
        elif len(stack) == 0 and i == ')': return False;
    return len(stack) == 0

File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    if(s[0] === ')' || s[s.length - 1] === '(') return false;
    
    const stack = [];
    for(let i of s){
        if(i === '(') stack.push(i);
        else stack.pop();
    }
    return stack.length === 0
}